jsonparse.ps1

```diff
@@ -0,0 +1,83 @@
+$location=$args[0]
+$repo=$args[1]
+$name=$args[2]
+
+$location
+$repo
+$name
+$repo.Title
+$JSON = (Get-Content $location) -join "`n" | ConvertFrom-Json
+
+$base = "https://github.com/"
+if($repo.Title -eq "github")
+{
+  $base = "https://github.com/"
+}
+#Renaming the key value of file
+$URI = $JSON.runs
+$JSON.runs.files.PSObject.Properties | foreach-object {
+  $URIReplacement = $_.Name
+  $content = $_.Value
+  $JSON.runs.files.PSObject.Properties.Remove($_.Name)
+  $URIReplacement = $base + $URIReplacement.SubString($URIReplacement.IndexOf($name))
+  $JSON.runs.files | add-member -Name $URIReplacement -Value $content -MemberType NoteProperty
+}
+
+$URI = $JSON.runs.files
```

Comment:
`runs` is an array, each element of which has a `files` property, so I'm not sure why this works. It's possible PowerShell's JSON handling automatically gives you the first array element if you don't specify an index, so that `$sarifLog.runs.files` is equivalent to `$sarifLog.runs[0].files`. You might look that up.

But in any event, the script should handle arrays with more than one element, and SARIF has many nested arrays, so you'll need nested loops. I'll give an example below.